backend/app/routers/runtime_proxy.py

```
@@ -188,21 +189,24 @@ async def proxy_request(
             )
         
-        # SECURITY: Derive workspace_id from authenticated context only
-        # NEVER trust workspace_id from headers that sandboxes can set
+        # Parse proxy auth header
         workspace_id = None
+        if x_proxy_auth:
+            try:
+                parts = x_proxy_auth.split(":", 1)
+                if len(parts) == 2:
+                    workspace_id = parts[1]
+            except Exception as e:
```

Ruff (0.14.3)

Do not catch blind exception: `Exception`

(BLE001)